transformer_engine/jax/csrc/extensions/attention.cpp

```diff
@@ -335,6 +333,124 @@ void FusedAttnForward(cudaStream_t stream, void **buffers, const char *opaque, s
+    int64_t dtype_, int64_t wkspace_dtype_, bool is_training, bool deterministic,
+    int64_t window_size_left, int64_t window_size_right) {
+  /* Descriptor data type conversion */
+  size_t input_batch = static_cast<size_t>(input_batch_);
+  size_t bias_batch = static_cast<size_t>(bias_batch_);
+  size_t q_max_seqlen = static_cast<size_t>(q_max_seqlen_);
+  size_t kv_max_seqlen = static_cast<size_t>(kv_max_seqlen_);
+  size_t attn_heads = static_cast<size_t>(attn_heads_);
+  size_t num_gqa_groups = static_cast<size_t>(num_gqa_groups_);
+  size_t bias_heads = static_cast<size_t>(bias_heads_);
+  size_t head_dim = static_cast<size_t>(head_dim_);
+  size_t max_segments_per_seq = static_cast<size_t>(max_segments_per_seq_);
+  size_t wkspace_size = static_cast<size_t>(wkspace_size_);
+  float scaling_factor = static_cast<float>(scaling_factor_);
+  float dropout_probability = static_cast<float>(dropout_probability_);
+  NVTE_Bias_Type bias_type = static_cast<NVTE_Bias_Type>(bias_type_);
+  NVTE_Mask_Type mask_type = static_cast<NVTE_Mask_Type>(mask_type_);
+  NVTE_QKV_Layout qkv_layout = static_cast<NVTE_QKV_Layout>(qkv_layout_);
+  DType dtype = static_cast<DType>(dtype_);
+  DType wkspace_dtype = static_cast<DType>(wkspace_dtype_);
+
+  /* Input buffers from XLA */
+  void *q = q_buf.untyped_data();
+  void *k = k_buf.untyped_data();
+  void *v = v_buf.untyped_data();
+  void *bias = bias_buf.untyped_data();
+  void *q_cu_seqlens = q_cu_seqlens_buf.untyped_data();
+  void *kv_cu_seqlens = kv_cu_seqlens_buf.untyped_data();
+  void *q_seq_offsets = q_seq_offsets_buf.untyped_data();
+  void *k_seq_offsets = k_seq_offsets_buf.untyped_data();
+  void *seed = seed_buf.untyped_data();
+
+  /* Output buffer from XLA */
+  void *output = output_buf->untyped_data();
+  void *softmax_aux = softmax_aux_buf->untyped_data();
+  void *rng_state = rng_state_buf->untyped_data();
+  void *workspace = workspace_buf->untyped_data();
+
+  /* Pack pointers and descriptor */
+  void *buffers[13] = {q,
+                       k,
+                       v,
+                       bias,
+                       q_cu_seqlens,
+                       kv_cu_seqlens,
+                       q_seq_offsets,
+                       k_seq_offsets,
+                       seed,
+                       output,
+                       softmax_aux,
+                       rng_state,
+                       workspace};
+  const CustomCallFusedAttnDescriptor descriptor = {
+      input_batch,   bias_batch,       q_max_seqlen,
+      kv_max_seqlen, attn_heads,       num_gqa_groups,
+      bias_heads,    head_dim,         max_segments_per_seq,
+      wkspace_size,  scaling_factor,   dropout_probability,
+      bias_type,     mask_type,        qkv_layout,
+      dtype,         wkspace_dtype,    is_training,
+      deterministic, window_size_left, window_size_right};
+
+  FusedAttnForwardImpl(stream, buffers, descriptor);
```

Comment:
Hi, I think we don't need to pack `buffers` and `descriptors` here to call the `FusedAttnForwardImpl`.
Since `FusedAttnForwardImpl` is only used for this `FusedAttnForwardFFI()` as `FusedAttnForward()` will be removed in the future, we can insert the computations into `FusedAttnForwardFFI` directly. 

When we will switch to new APIs with FFI for all custom calls, then all old custom call related functions will be removed, including the current `FusedAttnForward`.

Reply:
Revised as suggested